packages/api/src/modules/auth/services.ts

```diff
@@ -12,37 +12,47 @@ import {
   ISignInResponse,
 } from './types';
 import { LessThanOrEqual } from 'typeorm';
+import { PredicateService } from '../predicate/services';
 
 export class AuthService implements IAuthService {
   async signIn(payload: ICreateUserTokenPayload): Promise<ISignInResponse> {
-    return UserToken.create(payload)
-      .save()
-      .then(data => {
-        return {
-          accessToken: data.token,
-          avatar: data.user.avatar,
-          address: data.user.address,
-          user_id: data.user.id,
-          expired_at: data.expired_at,
-          workspace: {
-            id: data.workspace.id,
-            name: data.workspace.name,
-            avatar: data.workspace.avatar,
-            single: data.workspace.single,
-            permissions: data.workspace.permissions,
-          },
-          ...(data.user.type === 'WEB_AUTHN'
-            ? { webAuthn: data.user.webauthn }
-            : {}),
-        };
-      })
-      .catch(e => {
-        throw new Internal({
-          type: ErrorTypes.Internal,
-          title: 'Error on token creation',
-          detail: e,
-        });
+    try {
+      const data = await UserToken.create(payload).save();
+
+      let first_vault = undefined;
```

Comment:
All times we need create a new vault to user

i recomend add other column on user to include this first vault relation, because we need redirect user on login to this vault page